lib/cancancan/model_adapters/active_graph_adapter.rb

```diff
@@ -44,7 +44,11 @@ def self.property_matches?(subject, property, value)
         if subject.is_a?(ActiveGraph::Node::HasN::AssociationProxy)
           subject.where(property => value).exists?
         else
-          subject.send(property) == value
+          if value.is_a?(Enumerable)
+            subject.class.where(property => value, neo_id: subject.neo_id).exists?
```

Comment:
if this line is correct then it should handle both cases without the if statement. However, all 10 lines look dubious. If the subject is not a ..Proxy then one should be created rather than querying by neo_id.